server/src/db/repositories/reportRepository.js

```diff
@@ -77,12 +159,29 @@ const repository = () => {
   };
 
   const aggregateByProduct = (filters, callback) => {
-    ReportModel.aggregate(getLatestReportPipeline(filters)).exec(callback);
+    const aggregation = ReportModel.aggregate(getLatestReportPipeline(filters));
+
+    if (filters.validatedOffset !== null) {
```

Comment:
Can we change attribute names to be consistent with db naming conventions? In the "repository" we don't have information if offset is or is not valid, also it can be used in different context depending on business logic. I think skip and limit could be better names